src/test/java/org/dcsa/api_validator/ovs/v2/PostTimestampsTest.java

```diff
@@ -0,0 +1,588 @@
+        map.remove("vesselPosition");
+        map.remove("modeOfTransport");
+        map.remove("portCallServiceTypeCode");
+
+        map.put("eventLocation","aBCDaFGHaE"); // Wrong format
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                body(map).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(400);
+
+        map.put("eventLocation", "{}"); // Empty (fails Maybe - does not include mandatory parameters)
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                body(map).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(400);
+    }
+
+    @Test
+    public void testVesselPositionQueryParam() {
+
+        Map<String, String> map = (Map<String, String>) jsonToMap(VALID_TIMESTAMP);
+        map.remove("facilitySMDGCode");
+        map.remove("modeOfTransport");
+        map.remove("eventLocation");
+        map.remove("portCallServiceTypeCode");
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                header("testname","testVesselPositionQueryParam").
+                body(map).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(204);
+    }
+
+    // Test vesselPosition, fails as as format is JSON & (latitude & longitude required).
+    @Test
+    public void testVesselPositionFalseFormat() {
+        Map<String, String> map = (Map<String, String>) jsonToMap(VALID_TIMESTAMP);
+        map.remove("facilitySMDGCode");
+        map.remove("modeOfTransport");
+        map.remove("eventLocation");
+        map.remove("portCallServiceTypeCode");
+
+        map.put("vesselPosition","aBCDaFGHaE"); // Wrong format
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                body(map).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(400);
+
+        map.put("vesselPosition", "{}"); // Empty (fails - includes mandatory parameters)
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                body(map).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(400);
+    }
+
+    @Test
+    public void testModeOfTransportQueryParam() {
```

Comment:
I think this test name is wrong - as it is not testing query parameters. It is testing modeOfTransport field in a timestamp